pkg/gb/gameboy.go

```diff
@@ -306,6 +307,21 @@ func (gb *Gameboy) init(romFile string) error {
 	return nil
 }
 
+func (gb *Gameboy) initKeyHandlers() {
+	gb.keyHandlers = map[string]func(){
+		"Escape": gb.togglePaused,
+		"Equal":  changePallete,
+		"Q":      gb.Debug.toggleBackGround,
+		"W":      gb.Debug.toggleSprites,
+		"E":      gb.Debug.toggleOutputOpCode,
+		"D":      gb.printBGMap,
+		"7":      func() { gb.ToggleSoundChannel(1) },
+		"8":      func() { gb.ToggleSoundChannel(2) },
+		"9":      func() { gb.ToggleSoundChannel(3) },
+		"0":      func() { gb.ToggleSoundChannel(4) },
```

Comment:
It may be nicer for the io binding to define the key mappings instead of putting it into the `gb` package. That way if a future io binding isn't meant for a keyboard input it can define it's own keys. You could do this by creating some fake 'buttons' which are for the debugging cases (i.e. `ButtonPause`) and then you'd be able to make `keyHandlers` variable of type `map[Button]func()`? Or if you think that they're sufficiently different to emulated buttons you could create a new type like `DebugButtons` or `DebugKeys` in the `input.go` file and do similar.

It would involve repeating these debug shortcuts in each IO but I think that is an okay tradeoff.

Reply:
> It may be nicer for the io binding to define the key mappings instead of putting it into the gb package. (...) You could do this by creating some fake 'buttons' which are for the debugging cases (i.e. `ButtonPause`)

Yes, I agree. That would definitely be cleaner. 

> Or if you think that they're sufficiently different to emulated buttons you could create a new type like `DebugButtons` or `DebugKeys` in the `input.go` file and do similar.

I think having one type is nicer, because then we can list all possible `Button` values together and handle everything in one place. The field `KeyPressed` in `ButtonInput` can be removed then. Maybe `Button` should be renamed to `Input` then? Naming things is hard.

> It would involve repeating these debug shortcuts in each IO but I think that is an okay tradeoff.

I don't think there is an alternative, but I also think this repetition is not a bad thing: having a specific mapping from some input source to gameboy is exactly what I was going for.

Slightly off-topic: I think that the button input and the frame update output need to be split into different interfaces. Suppose I want to use a different input source (bot, usb controller, ...) and I still want to see the screen then I would have to copy and/or depend on the current pixelgl interface, which is pretty ugly IMO. I think this could be done in a future PR though, both because this one is pretty big already and because there is only IOBinding right now anyway.